esp/scm/ws_dfu.ecm

```diff
@@ -832,44 +832,83 @@ ESPresponse [exceptions_inline, nil_remove] EraseHistoryResponse
     ESParray<ESPStruct History, Origin> History;
 };
 
-ESPrequest DFUFileAccessRequest
+ESPStruct DFUFileAccessRequestBase
 {
-    string Name;
     string Cluster;
     int ExpirySeconds(60);
     ESPenum FileAccessRole AccessRole;
     ESPenum SecAccessType AccessType;
-    string JobId;
     bool ReturnJsonTypeInfo(false);
     bool ReturnBinTypeInfo(false);
 };
 
-// DFUPartLocations and DFUPartCopies part of DFUFileAccessResponse below
-ESPStruct DFUPartLocations
+ESPrequest DFUFileAccessRequest
+{
+    string Name;
+    string JobId;
+    EspStruct DFUFileAccessRequestBase RequestBase;
+};
+
+ESPStruct DFUFileCopy
 {
-    int LocationIndex;
-    string Location;
+    int CopyIndex;
+    string Host;
+    string Path;
 };
 
-ESPStruct DFUPartCopies
+ESPStruct DFUFilePart
 {
     int PartIndex;
-    ESParray<string> LocationIndexes;
+    ESParray<EspStruct DFUFileCopy> Copies;
 };
 
-ESPresponse [exceptions_inline] DFUFileAccessResponse
+ESPStruct DFUFileAccessInfo
 {
     string MetaInfoBlob;
     string ExpiryTime;
-    // {NumParts, FilePartLocations, FileParts} depend on ReturnFileInfo in request
+    // {NumParts, FileParts} depend on ReturnFileInfo in request
     int NumParts;        // number of parts in logical file
-    ESParray<EspStruct DFUPartLocations> FilePartLocations;
-    ESParray<EspStruct DFUPartCopies> FileParts;
+    ESParray<EspStruct DFUFilePart> FileParts;
     
     binary RecordTypeInfoBin;   // optional
     string RecordTypeInfoJson;  // optional
 };
 
+ESPresponse [exceptions_inline] DFUFileAccessResponse
+{
+    EspStruct DFUFileAccessInfo AccessInfo;
+};
+
+ESPrequest [nil_remove] DFUFileCreateRequest
+{
+    string Name;
+    string ECLRecordDefinition;
+    EspStruct DFUFileAccessRequestBase RequestBase;
+};
+
+ESPresponse [exceptions_inline, nil_remove] DFUFileCreateResponse
+{
+    string Name; //Name from request + '_' + Cluster from request + '.tmp'
```

Comment:
Should be renamed to something else, perhaps "CreateFileId".
Because although it _might_ looks like a name for now, it's something generated at the esp side and should be opaque to the client, who will simply pass back at publish time.